classes/submission/reviewAssignment/Collector.php

```diff
@@ -435,8 +445,24 @@ public function getQueryBuilder(): Builder
 
         $q->when(
             $this->reviewerIds !== null,
-            fn (Builder $q) =>
-            $q->whereIn('ra.reviewer_id', $this->reviewerIds)
+            fn (Builder $q) => $q
+                ->whereIn('ra.reviewer_id', $this->reviewerIds)
+                ->when(
+                    $this->isLastReviewRoundByReviewer,
+                    fn (Builder $q) => $q
+                        // Aggregate the latest review round number for the given assignment and reviewer
+                        ->leftJoinSub(
+                            DB::table('review_assignments as ramax')
+                                ->leftJoin('review_rounds as rrmax', 'ramax.review_round_id', '=', 'rrmax.review_round_id')
+                                ->select(['ramax.submission_id', 'ramax.reviewer_id'])
+                                ->selectRaw('MAX(rrmax.round) as latest_round')
+                                ->groupBy(['ramax.submission_id', 'ramax.reviewer_id']),
```

Comment:
I think this temporary sub table is calculated for all review assignments in the system, right?

While we need this only for given reviewerIds. So additionally adding WHERE condition for reviewerIds here I think could be good optimisation